builder/codebuild/codebuild.go

```diff
@@ -0,0 +1,245 @@
+package codebuild
+
+import (
+	"fmt"
+	"log"
+	"text/template"
+	"bytes"
+	"strings"
+	"io"
+	"os"
+	"errors"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/service/codebuild"
+	"github.com/remind101/conveyor/builder"
+	"golang.org/x/net/context"
+)
+
+const (
+	// Default Image for codebuild
+	DefaultCodebuildImage = "aws/codebuild/docker:1.12.1"
+
+	// Default AWS resource used by codebuild
+	DefaultCodebuildComputeType = "BUILD_GENERAL1_SMALL"
+
+)
+
+// Builder is a builder.Builder implementation that runs the build in a docker
+// container.
+type Builder struct {
+	codebuild *codebuild.CodeBuild
+
+	// Required field, arn of the instance-role required by codebuild
+	ServiceRole string
+
+	// The Image used by codebuild to build images. Defaults to 
+	// DefaultCodebuildImage
+	Image string
+
+	// The computing instances AWS CodeBuild will use. Defaults to
+	// DefaultCodebuildComputeType
+	ComputeType string
+
+	// Credentials for Dockerhub
+	DockerUsername string
+	DockerPassword string
+}
+
+type BuildSpecInput struct {
+	// Extend all the values already given in builder
+	*Builder
+
+	// The repository which is being built
+	Repository string
+
+	// The commit sha at which to do the build
+	Sha string
+}
+
+// NewBuilder returns a new Builder backed by the codebuild client.
+func NewBuilder(config client.ConfigProvider) *Builder {
+	return &Builder{
+		codebuild: codebuild.New(config),
+	}
+}
+
+// NewBuilderFromEnv returns a new Builder with a codebuild client
+// configured from the standard Docker environment variables.
+func NewBuilderFromEnv() (*Builder, error) {	
+
+	serviceRole := os.Getenv("CODEBUILD_SERVICE_ROLE")
+
+	if serviceRole == "" {
+		return nil, errors.New("CODEBUILD_SERVICE_ROLE must be set when using codebuild builder")
+	}
+
+	image := os.Getenv("CODEBUILD_IMAGE")
+
+	if image == "" {
+		image = DefaultCodebuildImage
+	}
+
+	computeType := os.Getenv("CODEBUILD_COMPUTE_TYPE")
+
+	if computeType == "" {
+		computeType = DefaultCodebuildComputeType
+	}
+
+	dockerUsername := os.Getenv("DOCKER_USERNAME")
```

Comment:
This is fine for testing, but we'll probably want to move these into [CLI flags](https://github.com/remind101/conveyor/blob/master/cmd/conveyor/worker.go#L16-L52)